ParseLoginUI/src/main/java/com/parse/ParseQueryRecyclerViewAdapter.java

```diff
@@ -0,0 +1,611 @@
+package com.parse;
+
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+import bolts.Capture;
+
+/**
+ * Created by lukask on 9/1/15.
+ */
+public class ParseQueryRecyclerViewAdapter<T extends ParseObject> extends RecyclerView.Adapter<ParseQueryRecyclerViewAdapter.ViewHolder> {
```

Comment:
We'll need both. In reality we should have named `ParseQueryAdapter` `ParseQueryListAdapter` since it's an implementation of `ListAdapter` and this could have been `ParseQueryAdapter`.

This class should actually go into `ParseUI-Widgets` under the package `com.parse.widget` so we could use `ParseQueryAdapter` name since it'll be in a different package...